app/javascript/mastodon/components/status/handled_link.tsx

```diff
@@ -84,27 +85,32 @@ export const HandledLink: FC<HandledLinkProps & ComponentProps<'a'>> = ({
 export const useElementHandledLink = ({
   hashtagAccountId,
   mentionAccountId,
+  mentions = [],
 }: {
   hashtagAccountId?: string;
   mentionAccountId?: string;
+  mentions?: ApiMentionJSON[];
```

Comment:
I'd prefer if there was just a single way to provide the `mentionAccountId`, e.g. allowing passing a function like `(element: HTMLAnchorElement) => string` to `mentionAccountId`. Then the `mentions.find` bit below could be left for the outer component to handle.